compiler/rustc_ty_utils/src/needs_drop.rs

```diff
@@ -274,6 +294,7 @@ fn drop_tys_helper<'tcx>(
     param_env: rustc_middle::ty::ParamEnv<'tcx>,
     adt_has_dtor: impl Fn(ty::AdtDef<'tcx>) -> Option<DtorType>,
     only_significant: bool,
+    exhaustive: bool,
```

Comment:
@nikomatsakis noted that `adt_significant_drop_tys` is dead code. I think that we should make `only_significant` "more accurate" and get rid of this additional `exhaustive` struct, since they kinda seem redundant.

Reply:
Actually, I do see the difference. So nevermind. but Ideally we gate them under a single bool for this API :)